express-notes-api/index.js

```diff
@@ -0,0 +1,140 @@
+const express = require('express');
+const fs = require('fs');
+const app = express();
+
+app.get('/api/notes', (req, res) => {
+  const noteArr = [];
+  fs.readFile('data.json', 'utf-8', (err, data) => {
+    if (err) {
+      console.error(err);
+      res.status(500).json({ error: 'Unexpected error!' });
+    }
+    const obj = JSON.parse(data);
+    const notes = obj.notes;
+    for (const key in notes) {
+      noteArr.push(notes[key]);
+    }
+    res.status(200).json(noteArr);
+  });
+});
+
+app.get('/api/notes/:id', (req, res) => {
+  const idNum = Number(req.params.id);
+  if (idNum < 0 || isNaN(idNum)) {
+    res.status(400).json({ error: 'That aint a positive number!' });
+  } else {
+    fs.readFile('data.json', (err, data) => {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ error: 'Unexpected error!' });
+      }
+      const obj = JSON.parse(data);
+      const notes = obj.notes;
+      if (notes[idNum]) {
+        res.status(200).json(notes[idNum]);
+      } else {
+        res.status(404).json({ error: `No entry found with id number ${idNum}` });
+      }
+    });
+  }
+});
+
+app.use(express.json());
+
+app.post('/api/notes', (req, res) => {
+  if (!req.body.content) {
+    res.status(400).json({ error: 'Content field required' });
+  } else {
+    fs.readFile('data.json', 'utf8', (err, data) => {
+      if (err) {
+        console.error(err);
+        res.status(500).json({ error: 'Unexpected error occured' });
+      }
```

Comment:
Be sure to `return` or include `else` to prevent the server from attempting to respond more than once to the same request.